### DISCUSSION TITLE
setting max_num_instances, max_detections_per_image

### DISCUSSION THREAD

**Discussion by Ekta246** [rwightman/efficientdet-pytorch#137] [General]
I still have this question, and I cannot find a perfect answer for the setting of max_num_instances and max_detections_per_image in the anchors.py

However, for now since I am working only on 1 class and 1 bounding box(only 1 GT label) per image I find the task easier with setting both the max points to 1.
I see my model now gives bbox with scores 98% and higher. I am confused if this going in correct direction?Or the max points should be increased(set higher than 1)?

I would really appreciate the answer to this question

TIA

**Comment by Ekta246** [comment]
> I still have this question, and I cannot find a perfect answer for the setting of max_num_instances and max_detections_per_image in the anchors.py
> 
> However, for now since I am working only on 1 class and 1 bounding box(only 1 GT label) per image I find the task easier with setting both the max points to 1.
> I see my model now gives bbox with scores 98% and higher. I am confused if this going in correct direction?Or the max points should be increased(set higher than 1)?
> 
> I would really appreciate the answer to this question
> 
> TIA

@MichaelMonashev I think you had asked the same question

**Reply by MichaelMonashev** [reply]
I just patched the code.

**Reply by jaideep11061982** [reply]
how one can get class wise confidence rather directly idx currently  in `output['detections][:,:,:,5]`?
@MichaelMonashev  @rwightman   any suggestions.
